src/telliot_feeds/utils/reporter_utils.py

```diff
@@ -177,5 +183,38 @@ def tkn_symbol(chain_id: int) -> str:
         return "XDAI"
     elif chain_id in ETHEREUM_CHAINS:
         return "ETH"
+    elif chain_id in FILECOIN_CHAINS:
+        return "FIL"
     else:
         return "Unknown native token"
+
+
+def fee_history_priority_fee_estimate(fee_history: FeeHistory, priority_fee_max: int) -> int:
+    """Estimate priority fee based on a percentile of the fee history.
+
+    Adapted from web3.py fee_utils.py
+
+    Args:
+        fee_history: Fee history object returned by web3.eth.fee_history
+        priority_fee_max: Maximum priority fee willing to pay
+
+    Returns:
+        Estimated priority fee in wei
+    """
+    priority_fee_min = 1_000_000_000  # 1 gwei
+    # grab only non-zero fees and average against only that list
+    non_empty_block_fees = [fee[0] for fee in fee_history["reward"] if fee[0] != 0]
+
+    # prevent division by zero in the extremely unlikely case that all fees within the polled fee
+    # history range for the specified percentile are 0
+    divisor = len(non_empty_block_fees) if len(non_empty_block_fees) != 0 else 1
+
+    priority_fee_average_for_percentile = Wei(round(sum(non_empty_block_fees) / divisor))
+
+    return (  # keep estimated priority fee within a max / min range
+        priority_fee_max
+        if priority_fee_average_for_percentile > priority_fee_max
```

Comment:
Is it just me or is this return kind of hard to read? Maybe something like this instead if you agree:
```py
if priority_fee_average_for_percentile > priority_fee_max:
        return priority_fee_max
elif priority_fee_average_for_percentile < priority_fee_min:
        return priority_fee_min
else:
        return priority_fee_average_for_percentile
 ```